src/Services/BookingService/BookingService.cs

```diff
@@ -70,38 +70,45 @@ public class BookingService : IBookingService
             _httpContextAccessor = httpContextAccessor;
         }
 
-        public async Task<BookingProcessEntity> Get(string baseUrl, Page currentPage, string guid)
+        public async Task<BookingProcessEntity> Get(string baseUrl, Page currentPage, string sessionGuid)
         {
             var bookingElement = currentPage.Elements
-                .Where(_ => _.Type == EElementType.Booking)
-                .FirstOrDefault();
+                .First(element => element.Type
+                    .Equals(EElementType.Booking));
 
-            var appointmentTimes = new List<AvailabilityDayResponse>();
+            List<AvailabilityDayResponse> appointmentTimes = new();
 
             var bookingInformationCacheKey = $"{bookingElement.Properties.QuestionId}{BookingConstants.APPOINTMENT_TYPE_SEARCH_RESULTS}";
-            var cachedAnswers = _distributedCache.GetString(guid);
 
-            if (cachedAnswers != null)
+            var appointmentType = bookingElement.Properties.AppointmentTypes
+                .GetAppointmentTypeForEnvironment(_environment.EnvironmentName);
+
+            var cachedAnswers = _distributedCache.GetString(sessionGuid);
+            FormAnswers convertedAnswers = new();
+            if (cachedAnswers is not null)
             {
-                var convertedAnswers = JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
+                convertedAnswers = JsonConvert.DeserializeObject<FormAnswers>(cachedAnswers);
+
+                if (appointmentType.NeedsMapping)
+                    _mappingService.MapAppointmentId(appointmentType, convertedAnswers);
 
                 if (convertedAnswers.FormData.ContainsKey(bookingInformationCacheKey))
                 {
-                    var cachedBookingInformation = JsonConvert.DeserializeObject<BookingInformation>(convertedAnswers.FormData[bookingInformationCacheKey].ToString());
-                    var cachedInfo = new List<object> { cachedBookingInformation };
-                    return new BookingProcessEntity { BookingInfo = cachedInfo };
+                    var cachedBookingInformation = JsonConvert
+                        .DeserializeObject<BookingInformation>(convertedAnswers.FormData[bookingInformationCacheKey].ToString());
+
+                    if (appointmentType.AppointmentId.Equals(cachedBookingInformation.AppointmentTypeId))
+                        return new BookingProcessEntity { BookingInfo = new() { cachedBookingInformation } };
                 }
```

Comment:
if the cachedAnswers is null, will the appointmentId get mapped? can only see the MapAppointmentId happenening within the cache if, would we not need to map if its not cached.

Reply:
this isnt an issue, there will be cacheanswers at this stage if mapping is required